src/routes/articles/create-article.e2e.test.js

```diff
@@ -0,0 +1,123 @@
+		 user = await fixtures.insertUser() 
+		const body = { id: user.id, email: user.email }
+		accessToken = fixtures.generateAccessToken(
+			{user : body}
+		)
+	})
+	describe('Failure', () => {
+		it('should return 400 if id is not a number', async () => {
+			const expectedError = {
+				"status" : "failed",
+				"error": {
+			  "message": "userId must be a number"
+				}
+		  }
+			return	fixtures.api()
+				.post(`/api/v1/articles`)
+				.set('Authorization', `Bearer ${accessToken}`)
+				.send({
+					userId : "userId",
+					title : faker.random.word(),
+					article: faker.random.words(),
+					published : faker.datatype.boolean(),
+					type : 'article'
+				})
+				.expect(400, expectedError)
+
+		})
+		it('should return 400 if title is invalid', async () => {
+			const expectedError = {
+				"status" : "failed",
+				"error": {
+			  "message": "title is not allowed to be empty"
+				}
+		  }
+			return	fixtures.api()
+				.post(`/api/v1/articles`)
+				.set('Authorization', `Bearer ${accessToken}`)
+				.send({
+					userId : user.id,
+					title : "",
+					article: faker.random.words(),
+					published : faker.datatype.boolean(),
+					type : 'article'
+				})
+				.expect(400, expectedError)
+
+		})
+		it('should return 400 if article is invalid', async () => {
+			const expectedError = {
+				"status" : "failed",
+				"error": {
+			  "message": "article is not allowed to be empty"
+				}
+		  }
+			return	fixtures.api()
+				.post(`/api/v1/articles`)
+				.set('Authorization', `Bearer ${accessToken}`)
+				.send({
+					userId : user.id,
+					title : faker.random.word(),
+					article: "",
+					published : faker.datatype.boolean(),
+					type : 'article'
+				})
+				.expect(400, expectedError)
+
+		})
+		it('should return 401 if request is not authenticated', async () => 
+			fixtures.api()
+				.post(`/api/v1/articles`)
+			    .expect(401))
+
+
+	})
+
+	describe('Success', () => {
+		let data
+		before(async ()=>{
+			data = {
+				userId : user.id,
+				title : faker.random.word(),
+				image : faker.image.imageUrl(),
+				article : faker.random.words(),
+				published : faker.datatype.boolean(),
+				type : 'article'
+			}
+		})
+		it('should return 201', async () => 
+			fixtures.api()
+				.post(`/api/v1/articles`)
+				.set('Authorization', `Bearer ${accessToken}`)
+				.send(data)
+				.expect(201)
+		)
+		it('should return the right response', async () =>
+			fixtures.api()
+				.post(`/api/v1/articles`)
+				.set('Authorization', `Bearer ${accessToken}`)
+				.send(data)
+				.expect(201)
+				.then((res) => {
+					expect(res.body).to.have.property('status', 'success')
+					expect(res.body).to.have.property('data')
+					expect(res.body.data.title).to.be.a('string')
+					expect(res.body.data.article).to.be.a('string')
+					expect(res.body.data.articleId).to.be.an('number')
+					expect(res.body.data.userId).to.be.a('number')
+					expect(res.body.data.message)
+						.to.eql('Article successfully posted')
```

Comment:
So our way of checking this now increases the likelihood of omitting certain fields. In this case, fields like `published` and `createdOn` don't appear

we also do not get to know if the right values of the fields are being returned. I suggest we go with something like this

In this suggestion, we check the autogenerated fields if they are part of the response since we cannot tell the value they come with

after that we take them out of the data and then we check the actual values of the rest of the data

```suggestion
					expect(res.body.data.articleId).to.be.an('number')
					expect(res.body.data.createdOn).to.be.a('string')

					// remove autogenerated data
					delete res.body.data.articleId
					delete res.body.data.createdOn

					expect(res.body).to.eql({
						status: 'success',
						data:{
							message: 'Article successfully posted',
							title: data.title,
							article: data.article,
							published: data.published,
							image: data.image,
							userId: user.id
						}						
					})
```